### PR TITLE
use eslint for ts and js

### PR DESCRIPTION
**Pull request by shoota**
# TL;DR

using eslint to lint ts with @typescript-eslint.
parse by it, and checking rules with configured the plugins features.

## Check this pr.

- passed lint each packages

### How to check

*   checkout this pr.
*   yarn 

#### check lint
*   yarn lint

#### check webpages
* yarn start
  * access to local docusaurus and check pages
  * access to local example site and check pages

#### check build and packing
* yarn build  

#### check lint
*   [x]  passed all lint

#### check webpages
* [x] it can see docusaurus pages normally
* [x] it can see example pages normally

#### check build and packing
* bundle  js files include License sign
* output js.map files

* [x] passed all CIs

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+module.exports = {
+  parser: '@typescript-eslint/parser',
```
Comment (shoota) [inline .eslintrc.js]: use @typescrpt-eslint and eslint plugins.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  plugins: ['import', '@typescript-eslint'],
+  rules: {
+    '@typescript-eslint/camelcase': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/prefer-interface': 'off',
+    '@typescript-eslint/no-use-before-define': 'off',
+    '@typescript-eslint/no-unused-vars': 'error',
+    '@typescript-eslint/no-explicit-any': 'off',
+    'no-return-assign': 0,
+    'no-console': [2, { allow: ["warn", "error"] }],
```
Comment (shoota) [inline .eslintrc.js]: agent logger will be logging crashed resonn by `warn`.

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    mocha: true,
+  },
+  plugins: ['import', '@typescript-eslint'],
+  rules: {
+    '@typescript-eslint/camelcase': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/prefer-interface': 'off',
+    '@typescript-eslint/no-use-before-define': 'off',
+    '@typescript-eslint/no-unused-vars': 'error',
+    '@typescript-eslint/no-explicit-any': 'off',
```
Comment (shoota) [inline .eslintrc.js]: explicit any is necessary now.
For example, custom `window` member is invalid in types. It need original declaration to fix it. (like as below)
```
const w = window as any
w.q = function......
```

File: website/i18n/en.json [hunk 4]
```diff
@@ -4,22 +4,51 @@
...
-    "linker-plugin": "Linker",
-    "optimizely_x-plugin": "Optimizely X",
-    "vwo-plugin": "Visual Website Optimizer",
-    "writing-plugins": "Plugins",
-    "Usage": "Usage",
-    "GitHub": "GitHub",
-    "Blog": "Blog",
-    "Plugins": "Plugins",
-    "Integrations": "Integrations"
+    "docs": {
```
Comment (shoota) [inline website/i18n/en.json]: updated by docusaurus commands.

File: package.json [hunk 5]
```diff
@@ -3,6 +3,9 @@
   "bugs": {
     "url": "https://github.com/userdive/agent.js/issues"
   },
+  "dependencies": {
+    "eslint-plugin-react": "7.13.0"
```
Comment (tatsushitoji) [inline package.json]: It should be `devDependencies` here instead of `dependencies` , shouldn't it?

Comment (shoota) [reply package.json]: oh...
I will fix it soon! thanks.
